### PR TITLE
fix(mongodb-memory-server-core): change import syntax and moved packa…

### PR DESCRIPTION
**Pull request by BigForNothing**
…ges to dependencies (#263)

## Related Issues

- #1

### CODE REVIEW

File: packages/mongodb-memory-server-core/src/MongoMemoryServer.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { ChildProcess } from 'child_process';
-import tmp from 'tmp';
+import * as tmp from 'tmp';
```
Comment (hasezoey) [inline packages/mongodb-memory-server-core/src/MongoMemoryServer.ts]: thanks didnt notice this when i was refactoring the file

File: packages/mongodb-memory-server-core/package.json [hunk 2]
```diff
@@ -31,6 +31,9 @@
   },
   "homepage": "https://github.com/nodkz/mongodb-memory-server",
   "devDependencies": {
+    "rimraf": "^3.0.0"
+  },
+  "dependencies": {
```
Comment (hasezoey) [inline packages/mongodb-memory-server-core/package.json]: like i said in #263, `@type` packages should never be in anything other than `devDependencies`

Comment (BigForNothing) [reply packages/mongodb-memory-server-core/package.json]: Please see my comments on #263 as Microsoft's TypeScript handbook states you should be including them in dependencies.

**Comment by hasezoey** [comment]
@BigForNothing ok, i understand now (sorry), but wouldnt it be better to put them into `peerDependencies`? because so it would not be an nested-install (like this package having its own version of some typings and some other package another version of typings)

**Comment by BigForNothing** [comment]
I would say no, because you are still requiring people to install it. Also, peerDependencies was created to [solve](https://nodejs.org/es/blog/npm/peer-dependencies/) the plugins problem (think eslint plugins).

**Comment by nodkz** [comment]
I always put `@types/*` packages to devDependencies. And I don't think that it's a good idea to install @types/* packages for a production build `npm install --production`. Anyway `mongodb-memory-server` package will be used as `devDependency` in most projects so there is no problem to put all `@types` packages to `dependencies`.

BUT damn, it's quite strange for me. I have a lot of other packages and had no problems.

So I asked Daniel (https://twitter.com/nodkz/status/1219160364014804992) about their recommendations in the handbook. I feel that they are incomplete or stale. Hope he answers soon.

**Comment by nodkz** [comment]
Crap! I removed my question to Daniel.

@BigForNothing you provide incorrect information to us.
If you read handbook recommendations carefully - they are about how to prepare `@types` packages NOT regular packages 😈

Please move `@types` to `devDependencies` back 😉

So your PR should have just one change and it will be enough:
```diff
- import tmp from 'tmp';
+ import * as tmp from 'tmp';
```

**Comment by hasezoey** [comment]
@nodkz just to note: i removed one import, the type for `DirResult` and replaced it with `tmp.DirResult` because tmp was already imported, and i didnt noticed that it was an *default* import and not what you mention now

**Comment by nodkz** [comment]
@BigForNothing please check the new version of the package and confirm that error in your env is gone.

Tnx ;)

**Comment by BigForNothing** [comment]
@nodkz I didn't actually. Did you go to the page that was linked from the issue thread? The example specifically shows a "regular" package with the name of "browserify-typescript-extension".

I think the confusion lies in that the very first section is about publishing declaration files.

Here is a [thread](https://github.com/microsoft/types-publisher/issues/81) where Daniel talks about this. While it's for MS's type publisher tool, it's in reference to why "--save" started showing up as the recommended install method instead of "--save-dev".

> Using --save-dev is fine when you're writing a simple application, and it won't be used as a library. The problem comes along when you might have dependencies. If you stored your type declarations in your devDependencies, your consumers would not automatically get your type declarations. They'd then have to manually add your declaration file dependencies as their devDependencies.

> Given that breaking consumers is a worse problem than slightly-larger packages, we've made --save the default in our documentation.

And im a reply further down: 

> My thought is dependencies since you're written as a library with dependents.

You can see that the intention would apply to this project, and "@types/tmp" should be in dependencies.

Im on mobile at the moment, so I'll test the latest changes when I'm on the computer.

**Pull request merged**